src/delta/delta.c

```diff
@@ -377,9 +394,9 @@ static int process_suffix(const char *suffix, const char *onlyprefix) {
 
         dropins = nulstr_contains(have_dropins, suffix);
 
-        top = ordered_hashmap_new(&string_hash_ops);
-        bottom = ordered_hashmap_new(&string_hash_ops);
-        drops = ordered_hashmap_new(&string_hash_ops);
+        top = ordered_hashmap_new(&enumerate_dir_hash_ops_free_free);
+        bottom = ordered_hashmap_new(&enumerate_dir_hash_ops_free_free);
+        drops = ordered_hashmap_new(&nested_hash_ops_free_free);
```

Comment:
Please switch `enumerate_dir` and friends to `ordered_hashmap_ensure_put()`.

Reply:
I had written it that way, but thought this way was cleaner.  I can change it.  Should I put the default &string_hash_ops back into the new calls?

Reply:
we generally want to get rid of explicit hash table allocation, and always use hashmap_ensure_put() to allocatethe hash tables on-the-fly on first inseration.

And if you want a string→string hash table where both key & value are freed, use the existing string_hash_ops_free_free hash_ops table. It's a common case, hence we have a generic one for that.